### PR TITLE
remove Visitor from Expression_toDt()

### PR DESCRIPTION
**Pull request by WalterBright**
Replaced it with a switch. Best reviewed with whitespace diffs turned off.

### CODE REVIEW

File: src/dmd/todt.d [hunk 1]
```diff
@@ -211,378 +211,383 @@ extern (C++) void Initializer_toDt(Initializer init, ref DtBuilder dtb)
...
+                    }
                 }
-
-                default:
-                    assert(0);
+                break;
             }
+
+            default:
+                printf("StringExp.toDt(type = %s)\n", e.type.toChars());
```
Comment (Basile-z) [inline src/dmd/todt.d]: maybe comment this ?

Comment (WalterBright) [reply src/dmd/todt.d]: This PR only does one thing, that's replace Visitor with switch.

Comment (Basile-z) [reply src/dmd/todt.d]: sorry i reviewed too fast and only with the diff.

Comment (Basile-z) [reply src/dmd/todt.d]: yeah you could leave it, my bad, I missed the assert(0) beyound.

**Comment by JinShil** [comment]
> Replaced it with a switch

What's the benefit?  It's quite a different pattern than the rest of DMD.

**Comment by WalterBright** [comment]
> What's the benefit?

1. Visitor is good when a tree is being traversed. That is not the case here, which is confusing because one has to carefully read it to see if it is being traversed or not.
2. Removing unnecessary dependencies on other classes.
3. Visitor cannot handle const arguments. Expression_toDt really should have const for e.
4. Remove complexity of setting up and tearing down the Visitor.

**Review by jacob-carlborg — APPROVED** [review]

**Comment by Basile-z** [comment]
@JinShil, there's been other visitors converted lately, in the same fashion.

**Review by ghost — APPROVED** [review]

**Pull request merged**